tensorflow_quantum/core/ops/tfq_utility_ops_test.py

```diff
@@ -126,5 +128,234 @@ def test_padded_to_ragged(self, padded_array):
+        #                             "present in the program."):
+        #     # symbol_names tensor has the right type, but invalid value.
+        #     tfq_utility_ops.tfq_resolve_parameters(
+        #         util.convert_to_tensor(circuit_batch), ['junk'],
+        #         symbol_values_array)
+
+        with self.assertRaisesRegex(TypeError, 'Cannot convert'):
+            # programs tensor has the wrong type.
+            tfq_utility_ops.tfq_resolve_parameters([1] * batch_size,
+                                                   symbol_names,
+                                                   symbol_values_array)
+
+        with self.assertRaisesRegex(TypeError, 'Cannot convert'):
+            # symbol_names tensor has the wrong type.
+            tfq_utility_ops.tfq_resolve_parameters(
+                util.convert_to_tensor(circuit_batch), [1], symbol_values_array)
+
+        with self.assertRaisesRegex(TypeError, 'Cannot convert'):
+            # symbol_values tensor has the wrong type.
+            tfq_utility_ops.tfq_resolve_parameters(
+                util.convert_to_tensor(circuit_batch), symbol_names,
+                [['junk']] * batch_size)
+
+        with self.assertRaisesRegex(TypeError, 'missing'):
+            # too few tensors.
+            # pylint: disable=no-value-for-parameter
+            tfq_utility_ops.tfq_resolve_parameters(
+                util.convert_to_tensor(circuit_batch), symbol_names)
+            # pylint: enable=no-value-for-parameter
+
+    def test_resolve_parameters_consistency_basic(self):
+        """Compare tfq op to cirq resolving."""
+        qubits = cirq.GridQubit.rect(1, 4)
+        circuit = cirq.Circuit()
+        symbols = []
+        for n, q in enumerate(qubits):
+            new_bit = sympy.Symbol("bit_{}".format(n))
+            circuit += cirq.X(q)**new_bit
+            symbols.append(new_bit)
+        symbol_names = [str(s) for s in symbols]
+
+        bitstring_list = [[0, 0, 0, 0], [0, 1, 0, 1], [1, 0, 1, 1]]
+        circuit_list = []
+        resolver_list = []
+        for bitstring in bitstring_list:
+            resolve_dict = {}
+            for s, b in zip(symbols, bitstring):
+                resolve_dict[s] = b
+            resolver_list.append(cirq.ParamResolver(resolve_dict))
+            circuit_list.append(circuit)
+
+        test_resolved_circuits = util.from_tensor(
+            tfq_utility_ops.tfq_resolve_parameters(
+                util.convert_to_tensor(circuit_list), symbol_names,
+                np.asarray(bitstring_list)))
+
+        expected_resolved_circuits = []
+        for circuit, resolver in zip(circuit_list, resolver_list):
+            expected_resolved_circuits.append(
+                cirq.resolve_parameters(circuit, resolver))
+
+        self.assertAllEqual(util.convert_to_tensor(expected_resolved_circuits),
+                            util.convert_to_tensor(test_resolved_circuits))
```

Comment:
Please compare them directly.

```python
self.assertAllEqual(expected_resolved_circuits, test_resolved_circuits)
```

Because protobuf doesn't guarantee that the serialized strings are encoded in the same order at every call.

Reply:
Changed to comparing Cirq circuits directly